### PR TITLE
Issue 17670 replace typography with text

### PR DESCRIPTION
**Pull request by ujwalkumar1995**
## Explanation

As part of this PR, I have replaced the usage of Typography with Text. The files are:

- ui/components/ui/numeric-input/numeric-input.component.js
- ui/components/app/signature-request/signature-request-message/signature-request-message.js
- ui/components/app/signature-request/signature-request.component.js

* Fixes #17670

### Before

 **ui/components/ui/numeric-input/numeric-input.component.js:**

Default:
Screenshot 2023-06-04 141849

With Detail:
Screenshot 2023-06-04 141911

With Error:
Screenshot 2023-06-04 141927

**ui/components/app/signature-request/signature-request-message/signature-request-message.js:**

Screenshot 2023-06-04 142000

**ui/components/app/signature-request/signature-request.component.js:**

Screenshot 2023-06-04 142151

### After

 **ui/components/ui/numeric-input/numeric-input.component.js:**

**Default:**
Screenshot 2023-06-04 141412

**With Detail:**
Screenshot 2023-06-04 141438

**With Error:**
Screenshot 2023-06-04 141519

**ui/components/app/signature-request/signature-request-message/signature-request-message.js:**

Screenshot 2023-06-04 141633

**ui/components/app/signature-request/signature-request.component.js:**

Screenshot 2023-06-04 141758

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by ujwalkumar1995** [comment]
I have read the CLA Document and I hereby sign the CLA

File: ui/components/app/signature-request/signature-request-message/signature-request-message.js [hunk 1]
```diff
@@ -79,14 +79,14 @@ export default function SignatureRequestMessage({
         className="signature-request-message__root"
         ref={setMessageRootRef}
       >
-        <Typography
-          fontWeight={FONT_WEIGHT.BOLD}
+        <Text
+          fontWeight={FontWeight.Bold}
           color={TextColor.textDefault}
           marginLeft={4}
           className="signature-request-message__title"
```
Comment (georgewrmarshall) [inline ui/components/app/signature-request/signature-request-message/signature-request-message.js]: I think we can remove the className and the styles associated styles now that we are using the `Text` component. Could you please remove the following css from `ui/components/app/signature-request/index.scss`

```  &__title {
    @include H5;

    font-weight: 500;
  }
```

```suggestion
```

File: ui/components/app/signature-request/signature-request.component.js [hunk 2]
```diff
@@ -324,26 +322,26 @@ export default class SignatureRequest extends PureComponent {
             />
           </div>
 
-          <Typography
+          <Text
             className="signature-request__content__title"
-            variant={TypographyVariant.H3}
+            variant={TextVariant.headingMd}
+            as="h3"
             fontWeight={FontWeight.Bold}
```
Comment (georgewrmarshall) [inline ui/components/app/signature-request/signature-request.component.js:328]: Shouldn't need the fontWeight as headingMd is already bold
```suggestion
```

File: ui/components/app/signature-request/signature-request.component.js [hunk 3]
```diff
@@ -13,10 +13,8 @@ import {
 import { MetaMetricsEventCategory } from '../../../../shared/constants/metametrics';
 import SiteOrigin from '../../ui/site-origin';
 import Button from '../../ui/button';
-import Typography from '../../ui/typography/typography';
 import ContractDetailsModal from '../modals/contract-details-modal/contract-details-modal';
 import {
-  TypographyVariant,
   FontWeight,
   TextAlign,
   TextColor,
```
Comment (georgewrmarshall) [inline ui/components/app/signature-request/signature-request.component.js:20]: Could you also please update the deprecated DISPLAY and BLOCK_SIZES in favour of enums

File: ui/components/app/signature-request-original/signature-request-original.component.js [hunk 4]
```diff
@@ -168,11 +168,7 @@ export default class SignatureRequestOriginal extends Component {
                 color={IconColor.infoDefault}
                 marginRight={2}
               />
-              <Text
-                variant={TextVariant.bodyXs}
-                color={TextColor.textDefault}
-                as="h7"
```
Comment (georgewrmarshall) [inline ui/components/app/signature-request-original/signature-request-original.component.js:174]: Removing as="h7" as there is no such thing as an h7 tag in html

File: ui/components/app/signature-request/index.scss [hunk 5]
```diff
@@ -47,12 +47,6 @@
   flex-direction: column;
   min-height: min-content;
 
-  &__title {
-    @include H5;
-
-    font-weight: 500;
-  }
-
```
Comment (georgewrmarshall) [inline ui/components/app/signature-request/index.scss:55]: Unneeded CSS

File: ui/components/app/signature-request/signature-request.stories.js [hunk 6]
```diff
@@ -79,6 +82,8 @@ DefaultStory.args = {
   fromAccount: MOCK_PRIMARY_IDENTITY,
   providerConfig: { name: 'Goerli ETH' },
   selectedAccount: MOCK_PRIMARY_IDENTITY,
+  hardwareWalletRequiresConnection: false,
+  currentCurrency: 'usd',
 };
```
Comment (georgewrmarshall) [inline ui/components/app/signature-request/signature-request.stories.js:87]: Adding some missing required props to clean up story console

**Review by georgewrmarshall — DISMISSED** [review]
Hey @ujwalkumar1995, this is looking great! I've made some small updates while I had your brach checked out on my local. Thanks for your contribution!

File: ui/components/app/signature-request/signature-request-message/signature-request-message.js [hunk 7]
```diff
@@ -3,10 +3,9 @@ import PropTypes from 'prop-types';
 import { debounce } from 'lodash';
 import { I18nContext } from '../../../../contexts/i18n';
 import Box from '../../../ui/box';
-import Typography from '../../../ui/typography';
+import { Text } from '../../../component-library';
 import {
   DISPLAY,
-  FONT_WEIGHT,
   FLEX_DIRECTION,
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/app/signature-request/signature-request-message/signature-request-message.js]: updated

File: ui/components/app/signature-request/signature-request-message/signature-request-message.js (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/app/signature-request/signature-request-message/signature-request-message.js]: updated

File: ui/components/app/signature-request/signature-request.component.js (hunk 2 above)
Comment (georgewrmarshall) [reply ui/components/app/signature-request/signature-request.component.js:328]: updated

File: ui/components/app/signature-request/signature-request.component.js (hunk 3 above)
Comment (georgewrmarshall) [reply ui/components/app/signature-request/signature-request.component.js:20]: updated

**Comment by ujwalkumar1995** [comment]
Thank you @georgewrmarshall for the feedback.
I will try to keep these things in mind while submitting future PRs.

**Review by georgewrmarshall — DISMISSED** [review]

**Review by garrettbear — DISMISSED** [review]
LGTM

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Review by garrettbear — APPROVED** [review]

**Pull request merged**